app/experimenter/nimbus-ui/src/components/App/ExperimentRoot/index.test.tsx

```diff
@@ -0,0 +1,309 @@
+    const slug = "experiment-demo-slug";
+    const { mock, experiment } = mockExperimentQuery(slug);
+
+    const SubjectInner = (props: RouteComponentProps) => {
+      const { slug, experiment } = useContext(ExperimentContext)!;
+      return (
+        <div data-testid="SubjectInner">
+          <ul>
+            <li data-testid="id">{experiment.id}</li>
+            <li data-testid="slug">{slug}</li>
+          </ul>
+        </div>
+      );
+    };
+
+    render(
+      <Subject {...{ slug, mocks: [mock] }}>
+        <SubjectInner path="edit" />
+      </Subject>,
+    );
+
+    await screen.findByTestId("SubjectInner");
+    expect(screen.getByTestId("id")).toHaveTextContent(String(experiment.id));
+    expect(screen.getByTestId("slug")).toHaveTextContent(slug);
+  });
+
+  const AnalysisSubjectInner = (props: RouteComponentProps) => {
+    const { experiment, analysis, analysisLoading, analysisRequired } =
+      useContext(ExperimentContext)!;
+    // Super-hacky way to track whether or not we ever rendered while
+    // analysis was loading.
+    const analysisLoadingRenders = useRef(0);
+    if (analysisLoading) analysisLoadingRenders.current++;
+    return (
+      <div data-testid="SubjectInner">
+        <ul>
+          <li data-testid="id">{experiment.id}</li>
+          <li data-testid="analysisLoading">{"" + analysisLoading}</li>
+          <li data-testid="analysisLoadingRenders">
+            {"" + analysisLoadingRenders.current}
+          </li>
+          <li data-testid="analysisRequired">{"" + analysisRequired}</li>
+          <li data-testid="analysis">{JSON.stringify(analysis)}</li>
+        </ul>
+      </div>
+    );
+  };
+
+  it("fetches analysis data for live experiments", async () => {
+    const slug = "experiment-demo-slug";
+    const { mock } = mockExperimentQuery(slug, {
+      status: NimbusExperimentStatusEnum.LIVE,
+    });
+
+    render(
+      <Subject {...{ slug, mocks: [mock] }}>
+        <AnalysisSubjectInner path="edit" />
+      </Subject>,
+    );
+
+    // Loading experiment...
+    expect(screen.getByTestId("page-loading")).toBeInTheDocument();
+    // Experiment loaded, fetching analysis...
+    await waitFor(() => {
+      expect(screen.queryByTestId("page-loading")).not.toBeInTheDocument();
+      expect(screen.getByTestId("analysisLoading")).toHaveTextContent("true");
+    });
+    // Analysis fetch initiated...
+    await waitFor(() => {
+      expect(fetchMock).toHaveBeenCalled();
+    });
+    // Analysis response received...
+    await waitFor(() => {
+      expect(screen.getByTestId("analysisLoading")).toHaveTextContent("false");
+      expect(screen.getByTestId("analysis")).toHaveTextContent(
+        MOCK_ANALYSIS_DATA,
+      );
+    });
+    // Should have rendered at least once while analysis was loading, because
+    // this component didn't require analysis
+    expect(screen.getByTestId("analysisLoadingRenders")).not.toHaveTextContent(
+      "0",
```

Comment:
It's a hack, could be done more cleanly if the hook was tested separately with `@testing-library/react-hooks`, since that tracks render history. But here it's easier just to count render iterations while asserting the rest of the expected steps

Reply:
Oh I see it's directly counting the renders.  Yep makes sense 👍